Fix server error when creating emails with empty subjects

kuzmany opened the pull request:
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✔️
| New feature/enhancement? (use the a.x branch)      | ❌
| Deprecations?                          | ❌
| BC breaks? (use the c.x branch)        | ❌
| Automated tests included?              | ✔️
| Related user documentation PR URL      | N/A
| Related developer documentation PR URL | N/A  
| Issue(s) addressed                     | Fixes #15394

## Description

This PR fixes a server error (500) that occurs when creating an email with a long internal name (>190 characters) and an empty subject field.

**The Issue:**
- PR #15147 introduced validation for email name length limits
- However, the `TextOnlyDynamicContentValidator` was not handling null/empty subject values properly
- When validation runs on a null subject, it throws `UnexpectedTypeException` instead of gracefully handling it

**The Fix:**
- Modified `TextOnlyDynamicContentValidator::validate()` to skip validation for null or empty values
- This prevents the server error while maintaining existing validation behavior for actual string content
- Added comprehensive functional test to prevent regression

**Changes Made:**
1. **Validator Fix** - Added null/empty value check in `TextOnlyDynamicContentValidator.php`
2. **Test Coverage** - Added `testLongNameWithEmptySubjectValidation()` functional test that verifies:
   - Long names with empty subjects don't cause 500 errors
   - Proper validation messages are returned instead
   - Server responds with client error (400) not server error (500)

---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally
2. Go to Channels → Emails
3. Create a new email
4. Enter a very long internal name (over 190 characters): `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa`
5. Leave the subject field completely empty
6. Try to save the email
7. **Expected Result**: Should show validation errors for both name length and required subject, NOT a 500 server error
8. Run the automated test: `ddev exec vendor/bin/phpunit app/bundles/EmailBundle/Tests/Controller/EmailControllerFunctionalTest.php::testLongNameWithEmptySubjectValidation`

**Before this fix:** 500 Internal Server Error with "Expected argument of type 'string', 'null' given"  
**After this fix:** Proper form validation with user-friendly error messages

kuzmany committed:
Fix TextOnlyDynamicContentValidator null value handling

Resolves issue where validation fails with UnexpectedTypeException
when subject field is null/empty while having a long email name.

- Add null/empty value check to skip validation for non-string values
- Prevents 500 error when creating emails with long names and empty subjects
- Maintains existing validation behavior for actual string content

Fixes #15394

kuzmany committed:
Add functional test for issue #15394

Test verifies that long email name with empty subject doesn't cause
500 server error but returns proper validation errors instead.

This test case specifically addresses the scenario described in issue
#15394 where TextOnlyDynamicContentValidator would throw an
UnexpectedTypeException when handling null subject values.

LeonOltmanns reviewed (approved):
Fixed the adressed issue - thanks for the quick work! :) 
<img width="293" height="360" alt="image" src="https://github.com/user-attachments/assets/9564a070-0bbd-4c17-a617-8a627ebc15fc" />

kuzmany committed:
Remove overly specific assertion from test

The 'This field is required' message might vary depending on
validation order or form implementation details. The key test
is that we get proper validation errors instead of a 500 error.

kuzmany committed:
Simplify test to rely on default empty subject behavior

Remove explicit empty string setting for subject field since
forms default to empty/null values when not set. This better
reflects the actual bug scenario from issue #15394.

kingsedem reviewed (approved):
It works as expected.
<img width="1918" height="775" alt="image" src="https://github.com/user-attachments/assets/6d825739-90f4-42f4-80e8-31c7e49ade49" />

andersonjeccel reviewed (approved):
It works

<img width="457" height="605" alt="image" src="https://github.com/user-attachments/assets/9b0e9860-aa9c-432f-ab28-80bd610b66af" />

Before:

<img width="1146" height="910" alt="image" src="https://github.com/user-attachments/assets/30549816-ed23-4195-a521-bab7d6744e52" />
